xdsl/irdl/operations.py

```diff
@@ -341,7 +344,10 @@ def __init__(
         self.constr = range_constr_coercion(attr)
 
 
-VarOpResult: TypeAlias = tuple[OpResult, ...]
+class VarOpResult(tuple[OpResult, ...]):
+    @property
+    def types(self):
+        return tuple(r.type for r in self)
```

Comment:
this is great if this doesn't need an `or [[]]`

Reply:
This wasn't actually needed where you used it :slightly_smiling_face: Happy to discuss it in chat if you're curious